source/ui/acquaman/AMGenericStepScanConfigurationView.cpp

```diff
@@ -483,10 +503,22 @@ void AMGenericStepScanConfigurationView::onDetectorSelectionChanged(QAbstractBut
 		AMDetector *detector = detectorButtonMap_.key(button, 0);
 
 		if (detector) {
-			if (button->isChecked())
+			if (button->isChecked()){
+
 				configuration_->addDetector(detector->toInfo());
-			else
+				i0ComboBox_->addItem(detector->name());
+			}
+
+			else {
+
 				configuration_->removeDetector(detector->toInfo());
+				i0ComboBox_->removeItem(i0ComboBox_->findText(detector->name()));
```

Comment:
Should there be some logic here as to what the i0 choice will be in the case that the detector which is currently the i0 is unchecked? I'm not 100% sure on the behvaiour of the QComboBox when the current selected item is removed, but I think it just selects the item with the previous index (if there is one) or following (if not). 

Might it be best to select "None" if the selected detector is removed, just so that it's consistent?